core/src/main/java/io/questdb/cairo/sql/PageFrameAddressCache.java

```diff
@@ -93,13 +106,25 @@ public void add(int frameIndex, @Transient PageFrame frame) {
 
         frameSizes.add(frame.getPartitionHi() - frame.getPartitionLo());
         frameFormats.add(frame.getFormat());
+        parquetAddresses.add(frame.getParquetAddr());
```

Comment:
can we do this when format it Parquet?

Reply:
Yes. That's because we mmap parquet files to avoid having to open the file per each worker. Instead, we mmap the file in table reader and use the start address + file size to initialize partition decoders on all worker threads.